### PR TITLE
simplify float::classify logic

### PR DESCRIPTION
**Pull request by RalfJung**
I played around with the float-classify test in the hope of triggering x87 bugs by strategically adding `black_box`, and still the exact expression @beetrees suggested [here](https://github.com/rust-lang/rust/pull/129835#issuecomment-2325661597) remains the only case I found where we get the wrong result on x87. Curiously, this bug only occurs when MIR optimizations are enabled -- probably the extra inlining that does is required for LLVM to hit the right "bad" case in the backend. But even for that case, it makes no difference whether `classify` is implemented in the simple bit-pattern-based version or the more complicated version we had before.

Without even a single testcase that can distinguish our `classify` from the naive version, I suggest we switch to the naive version.

r? @workingjubilee

### CODE REVIEW

File: tests/ui/numbers-arithmetic/issue-105626.rs
Comment (workingjubilee) [inline tests/ui/numbers-arithmetic/issue-105626.rs:1]: can we move this into the `float` dir with a better name? it has essentially nothing to do with integers except by dint of converting from them.

**Comment by RalfJung** [comment]
The base PR landed, so this is now ready.

**Comment by workingjubilee** [comment]
For the miscompile test,
> int-to-float-miscompile-issue-105626.rs

Can you split it into two revisions? One with `only-x86` and another with `ignore-x86`?

**Comment by RalfJung** [comment]
That test isn't even affected by this PR. It doesn't call `classify`, not even indirectly. I moved it around because you asked for it, otherwise it wouldn't even show up in the diff. I don't think I want to extend the scope of this PR to also cover "refactor all existing x87 tests".

**Comment by workingjubilee** [comment]
hmm, apologies, I must be going daffy...

you're right, there's just something chewing at my brain...

**Comment by workingjubilee** [comment]
I don't know why I'm overthinking this. If there's a problem revealed later on, and we deem fit, we can change the implementation again.

This is still liable to get "exciting" if rolled up however so let's not:

@bors r+ rollup=never

**Comment by workingjubilee** [comment]
> fatal: Fetched in submodule path 'src/llvm-project', but it did not contain 4b8d29c585687084bbcf21471e04f279d1eddc0a. Direct fetching of that commit failed.

This seems simply spurious?

@bors retry

**Pull request merged**